runtime/runtime-params-estimator/src/costs_to_runtime_config.rs

```diff
@@ -0,0 +1,152 @@
+use std::convert::TryFrom;
+
+use anyhow::Context;
+use near_primitives::runtime::fees::{
+    AccessKeyCreationConfig, ActionCreationConfig, DataReceiptCreationConfig, Fee,
+    RuntimeFeesConfig,
+};
+use near_primitives::types::Gas;
+use near_vm_logic::{ExtCostsConfig, VMConfig, VMLimitConfig};
+use node_runtime::config::RuntimeConfig;
+
+use crate::cost::Cost;
+use crate::cost_table::CostTable;
+
+/// Turn a [`CostTable`] into a [`RuntimeConfig`].
+///
+/// Will fail if [`CostTable`] doesn't contain all costs.
+///
+/// Note that the actual [`RuntimeConfig`] we use is currently hard-coded -- we
+/// don't really use this function in production.
+pub fn costs_to_runtime_config(cost_table: &CostTable) -> anyhow::Result<RuntimeConfig> {
+    let regular_op_cost = cost_table
+        .get(Cost::WasmInstruction)
+        .with_context(|| format!("undefined cost: {}", Cost::WasmInstruction))?;
+
+    let res = RuntimeConfig {
+        wasm_config: VMConfig {
+            ext_costs: ext_costs_config(cost_table)?,
+            grow_mem_cost: 1,
+            regular_op_cost: u32::try_from(regular_op_cost).unwrap(),
+            limit_config: VMLimitConfig::default(),
+        },
+        transaction_costs: runtime_fees_config(cost_table)?,
+        ..RuntimeConfig::default()
+    };
+    Ok(res)
+}
+
+fn runtime_fees_config(cost_table: &CostTable) -> anyhow::Result<RuntimeFeesConfig> {
+    let fee = |cost: Cost| -> anyhow::Result<Fee> {
+        let total_gas =
+            cost_table.get(cost).with_context(|| format!("undefined cost: {}", cost))?;
+        // Split the total cost evenly between send and execution fee.
```

Comment:
I was always curious if this logic is correct.

Reply:
The current formula seems correct to me: we split the cost in half, one for execution, one for sending (and sending can be cross-shard or same shard). What is not cleart to me is that why we have two different send fees. It seems that we want `send_sir + execution = total_gas` and `send_not_sir + execution = total_gas` to hold, which means that both sends should be equal.